### PR TITLE
Fix some Python 3 incompatibilities and issues with str/bytes conversion

### PR DESCRIPTION
**Pull request by SpotlightKid**
Tentative solution for #8

### CODE REVIEW

**Comment by SpotlightKid** [comment]
With this change I can use walrus.model in Python3, but I'm not entirely sure this is the best way to fix the issue in `models.Query.get_primary_hash_key()`. With this patch, `models.TextField.db_value` doesn't encode Python 3 (unicode) strings, but relies on redis-py to do so.

Another approach would be to not to use `db_value` in `get_primary_hash_key` but explicitly convert the primary key value to a string.

File: walrus/database.py [hunk 1]
```diff
@@ -53,6 +53,7 @@ def __init__(self, *args, **kwargs):
             scripts.
         """
         script_dir = kwargs.pop('script_dir', None)
+        self.encoding = kwargs.get('encoding') or 'utf-8'
```
Comment (coleifer) [inline walrus/database.py]: It looks like redis-py already supports a `charset` parameter:

https://redis-py.readthedocs.org/en/latest/index.html#redis.Redis

Perhaps that should be used instead?

**Comment by coleifer** [comment]
Is c104153, in combination with `decode_responses=True`, sufficient to resolve the problem?

File: walrus/database.py (hunk 1 above)
Comment (SpotlightKid) [inline walrus/database.py]: `charset` ist deprecated and replaced by `encoding`.

**Comment by SpotlightKid** [comment]
I'll have to check, but it seems to me that the `value.encode()` would still cause a problem when generating the key in `Query`.

**Pull request closed without merging**